pkg/env/env.go

```diff
@@ -367,12 +367,16 @@ func (e *testEnv) execFeature(ctx context.Context, t *testing.T, f types.Feature
 	return ctx
 }
 
-// featureInfoFromFeature just copies the values from the Feature but, as a new type,
-// it has less risk of abuse and side-effects. Meant to solely be informational.
-func featureInfoFromFeature(f types.Feature) types.FeatureInfo {
-	return types.FeatureInfo{
-		Name:   f.Name(),
-		Labels: f.Labels(),
-		Steps:  f.Steps(),
+// deepCopyFeature just copies the values from the Feature but creates a deep
+// copy to avoid mutation when we just want an informational copy.
+func deepCopyFeature(f types.Feature) types.Feature {
+	fcopy := features.New(f.Name())
+	for k, v := range f.Labels() {
+		fcopy = fcopy.WithLabel(k, v)
 	}
+	f.Steps()
+	for _, step := range f.Steps() {
+		fcopy = fcopy.WithStep(step.Name(), step.Level(), nil)
```

Comment:
It looks like `fcopy.WithStep` is copying the original step functions.  I think that is what should be avoided.  Maybe instead of a generic deepCopyFeature function, the function should be `deepCopyFeatureInfo` still returns a feature but does not copy the steps.

Reply:
No, the last value there is the function and I'm specifically passing in `nil` instead of `step.Func()`